### PR TITLE
Fix TypeScript error to export `Plugin` and `RuleContext`

### PR DESCRIPTION
**Pull request by henryruhs**
Export types for plugin authoring.

Related to https://github.com/stylelint/stylelint/issues/6662

### CODE REVIEW

**Review by ybiquitous — CHANGES_REQUESTED** [review]
@henryruhs Thanks for the pull request. `npm run lint:types` fails. Could you please check it?

I'll add a changeset later; no worries.

**Comment by henryruhs** [comment]
I fixed it, sorry for that

File: types/stylelint/index.d.ts [hunk 1]
```diff
@@ -9,7 +9,9 @@ declare module 'stylelint' {
 	export type Severity = 'warning' | 'error';
 
 	type ConfigExtends = string | string[];
-	type Plugin = { default?: { ruleName: string; rule: Rule } } | { ruleName: string; rule: Rule };
+	export type Plugin =
```
Comment (ybiquitous) [inline types/stylelint/index.d.ts:12]: [suggest] Can you also add JSDOC, please?

```suggestion
	/**
	 * A Stylelint plugin.
	 */
	export type Plugin =
```

File: types/stylelint/index.d.ts [hunk 2]
```diff
@@ -180,7 +182,7 @@ declare module 'stylelint' {
 		optional?: boolean;
 	};
 
-	type RuleContext = {
+	export type RuleContext = {
```
Comment (ybiquitous) [inline types/stylelint/index.d.ts:185]: [suggest] Also here:

```suggestion
	/**
	 * A rule context.
	 */
	export type RuleContext = {
```

File: types/stylelint/index.d.ts [hunk 3]
```diff
@@ -620,6 +622,8 @@ declare module 'stylelint' {
 		};
 	};
 
+	export type Stylelint = Omit<PublicApi, '_createLinter'>;
+
```
Comment (ybiquitous) [inline types/stylelint/index.d.ts:626]: [suggest] Let's remove the new type to minimize this pull request change. Perhaps, later refactoring PRs may add similar types.

```suggestion
```

**Review by ybiquitous — CHANGES_REQUESTED** [review]
Thanks for fixing the errors. I've left a few minor suggestions. Please consider addressing them.

**Comment by henryruhs** [comment]
Requested changes have been applied

**Review by ybiquitous — APPROVED** [review]
Thank you! LGTM 👍🏼

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**